clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -336,16 +337,28 @@ func (p *pingChecker) run() {
 				time.Sleep(pingBackoff)
 				continue
 			}
+			var ips []string
+			var port int32
 			switch p.svc.Spec.Type {
 			case corev1.ServiceTypeClusterIP:
-				cmd := fmt.Sprintf("curl %s:%d", p.svc.Spec.ClusterIP, p.svc.Spec.Ports[0].Port)
-				_, err = execservice.RunCommand(pod, cmd)
+				ips = p.svc.Spec.ClusterIPs
+				port = p.svc.Spec.Ports[0].Port
 			case corev1.ServiceTypeNodePort:
-				cmd := fmt.Sprintf("curl %s:%d", pod.Status.HostIP, p.svc.Spec.Ports[0].NodePort)
-				_, err = execservice.RunCommand(pod, cmd)
+				ips = []string{pod.Status.HostIP}
+				port = p.svc.Spec.Ports[0].NodePort
 			case corev1.ServiceTypeLoadBalancer:
-				cmd := fmt.Sprintf("curl %s:%d", p.svc.Status.LoadBalancer.Ingress[0].IP, p.svc.Spec.Ports[0].Port)
-				_, err = execservice.RunCommand(pod, cmd)
+				for _, ingress := range p.svc.Status.LoadBalancer.Ingress {
+					ips = append(ips, ingress.IP)
+				}
+				port = p.svc.Spec.Ports[0].Port
+			}
+			for _, ip := range ips {
+				address := net.JoinHostPort(ip, fmt.Sprint(port))
+				command := fmt.Sprintf("curl %s", address)
+				_, err = execservice.RunCommand(pod, command)
+				if err != nil {
```

Comment:
Consider using [ErrorList](https://github.com/kubernetes/perf-tests/blob/4a4b12023fe092bd4bfd69762c356bcc2f776667/clusterloader2/pkg/errors/error_list.go#L24-L28) for collecting errors from attempted connections to all the IP addresses rather than immediately breaking from the loop on receiving the first non-nil error here.

Reply:
I agree that would be useful, if we wanted to return errors, or log them

But here we just want to check if both IPs are reachable, what is the benefit of checking second one, if we know, that first one already failed?

Reply:
Ah, I didn't read the rest of this measurement's logic. I thought we're plugging this error through somewhere higher up the call stack and wanted to know if, for instance, IPv6 network programming failed but IPv4 did not  🤷.

In that case I'm okay with leaving it as it is.